### ISSUE TITLE
Akka HTTP fails with A Merge must have more than 1 input port

### ISSUE THREAD

**Issue by relgames** [akka/akka#19549]
This sample https://gist.github.com/relgames/38aac67d4020d7a95f37 works fine with Akka HTTP 2.0-M2 but fails with 2.0 and 2.0.2:

```
[2016-01-20 21:20:52,080] [  DEBUG] [spatcher-3] [akka://default/user/PoolInterfaceActor-0] a.h.i.e.c.PoolInterfaceActor: (Re-)starting host connection pool to 127.0.0.1:8082
[2016-01-20 21:20:52,187] [  ERROR] [spatcher-3] [akka://default/user/PoolInterfaceActor-0] a.a.OneForOneStrategy: requirement failed: A Merge must have more than 1 input port
akka.actor.ActorInitializationException: exception during creation
	at akka.actor.ActorInitializationException$.apply(Actor.scala:172) ~[akka-actor_2.11-2.4.0.jar:na]
	at akka.actor.ActorCell.create(ActorCell.scala:605) ~[akka-actor_2.11-2.4.0.jar:na]
	at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:460) ~[akka-actor_2.11-2.4.0.jar:na]
	at akka.actor.ActorCell.systemInvoke(ActorCell.scala:482) ~[akka-actor_2.11-2.4.0.jar:na]
	at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:282) [akka-actor_2.11-2.4.0.jar:na]
	at akka.dispatch.Mailbox.run(Mailbox.scala:223) [akka-actor_2.11-2.4.0.jar:na]
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234) [akka-actor_2.11-2.4.0.jar:na]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [scala-library-2.11.7.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [scala-library-2.11.7.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [scala-library-2.11.7.jar:na]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [scala-library-2.11.7.jar:na]
Caused by: java.lang.IllegalArgumentException: requirement failed: A Merge must have more than 1 input port
	at scala.Predef$.require(Predef.scala:219) ~[scala-library-2.11.7.jar:na]
	at akka.stream.scaladsl.Merge.<init>(Graph.scala:40) ~[akka-stream-experimental_2.11-2.0.2.jar:na]
	at akka.stream.scaladsl.Merge$.apply(Graph.scala:23) ~[akka-stream-experimental_2.11-2.0.2.jar:na]
	at akka.http.impl.engine.client.PoolFlow$$anonfun$apply$1.apply(PoolFlow.scala:81) ~[akka-http-core-experimental_2.11-2.0.2.jar:na]
	at akka.http.impl.engine.client.PoolFlow$$anonfun$apply$1.apply(PoolFlow.scala:73) ~[akka-http-core-experimental_2.11-2.0.2.jar:na]
	at akka.stream.scaladsl.GraphApply$class.create(GraphApply.scala:16) ~[akka-stream-experimental_2.11-2.0.2.jar:na]
	at akka.stream.scaladsl.GraphDSL$.create(Graph.scala:707) ~[akka-stream-experimental_2.11-2.0.2.jar:na]
	at akka.http.impl.engine.client.PoolFlow$.apply(PoolFlow.scala:73) ~[akka-http-core-experimental_2.11-2.0.2.jar:na]
	at akka.http.impl.engine.client.PoolInterfaceActor.initConnectionFlow(PoolInterfaceActor.scala:71) ~[akka-http-core-experimental_2.11-2.0.2.jar:na]
	at akka.http.impl.engine.client.PoolInterfaceActor.<init>(PoolInterfaceActor.scala:59) ~[akka-http-core-experimental_2.11-2.0.2.jar:na]
	at akka.http.impl.engine.client.PoolGateway$$anonfun$1.apply(PoolGateway.scala:43) ~[akka-http-core-experimental_2.11-2.0.2.jar:na]
	at akka.http.impl.engine.client.PoolGateway$$anonfun$1.apply(PoolGateway.scala:43) ~[akka-http-core-experimental_2.11-2.0.2.jar:na]
	at akka.actor.TypedCreatorFunctionConsumer.produce(Props.scala:339) ~[akka-actor_2.11-2.4.0.jar:na]
	at akka.actor.Props.newActor(Props.scala:259) ~[akka-actor_2.11-2.4.0.jar:na]
	at akka.actor.ActorCell.newActor(ActorCell.scala:561) ~[akka-actor_2.11-2.4.0.jar:na]
	at akka.actor.ActorCell.create(ActorCell.scala:587) ~[akka-actor_2.11-2.4.0.jar:na]
	... 9 common frames omitted
```

I tried with Http.get(system).<Integer>cachedHostConnectionPool(...) but it fails as well.

**Comment by ktoso** [comment]
Related to: https://github.com/akka/akka/issues/19257

**Comment by relgames** [comment]
@ktoso - but I don't use Broadcast. I use Http.get(system).cachedHostConnectionPool(...) as described in the documentation. It used to work in 2.0-M2. 
Are you saying cachedHostConnectionPool can't be used with a stream of just 1 element?

**Comment by relgames** [comment]
Hmm it fails even with Http.get(actorSystem).singleRequest(...)
Strange it got into 2.0 and 2.0.2 releases. Those are basic functions, aren't there unit tests for that?
Am I using it correctly? I'm following documentation.

**Comment by ktoso** [comment]
It's not about number of elements in the stream, but number of fan-in fan-out ports, and yes that limitation was added recently and is the same for Broadcast as it is for Merge, see the linked ticket for details.

I'll dive into the example after a good nights sleep and give more details :) Thanks for reporting

**Comment by ktoso** [comment]
Error might have slipped through in the Java layer... will research tomorrow.

**Comment by relgames** [comment]
Thanks!! Here is an example with cachedHostConnectionPool https://gist.github.com/relgames/0c2005bae42922ab2da3

**Comment by johanandren** [comment]
Example code does not repeat error, so I guess there must be configuration setting the max connections to 1.

Anyways, I will fix Merge to allow just one input, (and the related Broadcast with just one output as well #19257 )

**Comment by ktoso** [comment]
Yeah I was thinking just that – can you share your config as well @relgames ?

**Comment by relgames** [comment]
I actually have 1024 in the config file:
```
akka {
  loglevel = "DEBUG"
  loggers = ["akka.event.slf4j.Slf4jLogger"]
  logging-filter = "akka.event.slf4j.Slf4jLoggingFilter"
  log-dead-letters = 10
  log-dead-letters-during-shutdown = on

  http {
    host-connection-pool {
      max-connections = 1
      max-open-requests = 1024
      pipelining-limit = 1
    }
  }

}
```

**Comment by ktoso** [comment]
yep, the max-connections is the culprit here – see here https://github.com/akka/akka/blob/stream-http-2.0/akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolFlow.scala#L81 
May I ask why you want a max of 1 connection there?

**Comment by relgames** [comment]
That is a setting left when I was trying to find a solution for another issue (here it is if anyone is interested http://stackoverflow.com/questions/34808752/how-to-properly-call-a-single-server-from-multiple-actors-web-handlers-using-a/)

**Comment by drewhk** [comment]
the fix to be backported here is #19257
